hadoop-tools/hadoop-azure/dev-support/testrun-scripts/configsupport.sh

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FILE=$1
+
+if [ ! -f "$FILE" ]; then
```

Comment:
One overall suggestion would be to break down the different operations into functions, as we have lot going into the script. Some suggestion would be like 
check_dependencies() { ... }
validate_input_file() { ... }
prompt_namespace_type() { ... }
rename_configurations() { ... }
remove_obsolete_configs() { ... }
handle_defaultFS_endpoint() { ... }